device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -119,22 +114,22 @@
         this.receivePeriodInMilliseconds = receivePeriodInMilliseconds;
 
         /* Codes_SRS_DEVICE_IO_21_006: [The constructor shall set the `state` as `DISCONNECTED`.] */
-        this.state = IotHubClientState.CLOSED;
+        this.state = IotHubConnectionStatus.DISCONNECTED;
 
         if (protocol == IotHubClientProtocol.AMQPS_WS || protocol == IotHubClientProtocol.MQTT_WS)
         {
             this.config.setUseWebsocket(true);
         }
 
-        this.transport = new IotHubTransport(config);
+        this.transport = new IotHubTransport(config, this);
```

Comment:
Passing ```this``` in as a parameter so that the transpor layer can callback up to this layer when the connection status changes